Classes/BITMetricsManager.m

```diff
@@ -74,13 +74,12 @@ - (void)startManager {
 
 - (void)setDisabled:(BOOL)disabled {
   if (_disabled == disabled) { return; }
-  
+    _disabled = disabled;
   if (disabled) {
     [self unregisterObservers];
   } else {
-    [self registerObservers];
+    [self startManager];
```

Comment:
We might even not have a session id, if startManager was not called, and since we a starting new session each time app is in background mode for 20 seconds or more, i don't think we need to worry about starting new session here. Besides it is rather logical to start new session each time after enabling metrics manager.